clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -1024,13 +1023,15 @@ protected void leaderExecuteTopicSwitch(
    */
   void prepareOffsetCheckpointAndStartConsumptionAsLeader(
       PubSubTopic leaderTopic,
-      PartitionConsumptionState partitionConsumptionState) {
+      PartitionConsumptionState partitionConsumptionState,
+      boolean isTransition) {
     Set<String> leaderSourceKafkaURLs = getConsumptionSourceKafkaAddress(partitionConsumptionState);
     if (leaderSourceKafkaURLs.size() != 1) {
       throw new VeniceException("In L/F mode, expect only one leader source Kafka URL. Got: " + leaderSourceKafkaURLs);
     }
+    boolean useLcro = isTransition && isGlobalRtDivEnabled();
     long upstreamStartOffset = partitionConsumptionState
-        .getLeaderOffset(OffsetRecord.NON_AA_REPLICATION_UPSTREAM_OFFSET_MAP_KEY, pubSubTopicRepository);
+        .getLeaderOffset(OffsetRecord.NON_AA_REPLICATION_UPSTREAM_OFFSET_MAP_KEY, pubSubTopicRepository, useLcro);
```

Comment:
`public static final String NON_AA_REPLICATION_UPSTREAM_OFFSET_MAP_KEY = "";`

So, `NON_AA_REPLICATION_UPSTREAM_OFFSET_MAP_KEY` is always an empty string, can it cause some trouble  for us, since `getLatestConsumedRtOffset` always returns `0` when it couldn't find anything from the map?

Let's add some comments and a TODO, if this is an issue.

Reply:
This is the legacy issue. If we move away from using it right now, it will cause trouble as it won't load the pre-stored state. We can definitely do a migration and dual-write it, but I'd suggest let's not do it in the same PR